runtime/runtime_impl_jetty12/src/main/java/com/google/apphosting/runtime/jetty/JettyServletEngineAdapter.java

```diff
@@ -118,27 +120,45 @@ public InvocationType getInvocationType() {
             return InvocationType.BLOCKING;
           }
         };
-    rpcConnector =
-        new DelegateConnector(server, "RPC") {
-          @Override
-          public void run(Runnable runnable) {
-            // Override this so that it does the initial run in the same thread.
-            // Currently, we block until completion in serviceRequest() so no point starting new
-            // thread.
-            runnable.run();
-          }
-        };
-    server.addConnector(rpcConnector);
+
+    // Don't add the RPC Connector if in HttpConnector mode.
+    if (!isHttpConnectorMode)
+    {
+      rpcConnector =
+          new DelegateConnector(server, "RPC") {
+            @Override
+            public void run(Runnable runnable) {
+              // Override this so that it does the initial run in the same thread.
+              // Currently, we block until completion in serviceRequest() so no point starting new
+              // thread.
+              runnable.run();
+            }
+          };
+
+      HttpConfiguration httpConfiguration = rpcConnector.getHttpConfiguration();
+      httpConfiguration.setSendDateHeader(false);
+      httpConfiguration.setSendServerVersion(false);
+      httpConfiguration.setSendXPoweredBy(false);
+      if (LEGACY_MODE) {
```

Comment:
Just a question there (hard to diff), LEGACY_MODE is still in both rpc and http connector, right?

Reply:
Yes but for http connector it is done in `com.google.apphosting.runtime.jetty.proxy.JettyHttpProxy#newConnector`.